modules/touch/touchmodule.cpp

```diff
@@ -120,50 +115,93 @@ bool TouchModule::processNewInput() {
     }
 }
 
-void TouchModule::processNewWebInput(const std::vector<TuioCursor>& listOfContactPoints) {
-    bool isWebPositionCallbackZero =
-        (_webPositionCallback.x == 0 && _webPositionCallback.y == 0);
-    bool isSingleContactPoint = (listOfContactPoints.size() == 1);
-    if (isSingleContactPoint && isWebPositionCallbackZero) {
-        glm::ivec2 res = global::windowDelegate.currentWindowSize();
-        glm::dvec2 pos = glm::vec2(
-            listOfContactPoints.at(0).getScreenX(res.x),
-            listOfContactPoints.at(0).getScreenY(res.y)
-        );
-
-#ifdef OPENSPACE_MODULE_WEBBROWSER_ENABLED
-        WebBrowserModule& module = *(global::moduleEngine.module<WebBrowserModule>());
-        if (module.eventHandler().hasContentCallback(pos.x, pos.y)) {
-            _webPositionCallback = glm::vec2(pos.x, pos.y);
-            module.eventHandler().touchPressCallback(pos.x, pos.y);
+void TouchModule::clearInputs() {
+    for(const auto& input : _deferredRemovals) {
+        for(TouchInputs& points : _touchPoints) {
+            if(points.getFingerId() == input.fingerId
+                && points.getTouchDeviceId() == input.touchDeviceId) {
+                points = std::move(_touchPoints.back());
+                _touchPoints.pop_back();
+                break;
+            }
         }
     }
-    // Send mouse release if not same point input
-    else if (!isSingleContactPoint && !isWebPositionCallbackZero) {
-        WebBrowserModule& module = *(global::moduleEngine.module<WebBrowserModule>());
-        module.eventHandler().touchReleaseCallback(_webPositionCallback.x,
-            _webPositionCallback.y);
-        _webPositionCallback = glm::vec2(0, 0);
-#endif
+    _deferredRemovals.clear();
+}
+
+bool TouchModule::addTouchInput(TouchInput input) {
+    _touchPoints.emplace_back(input);
+    return true;
+}
+
+bool TouchModule::updateOrAddTouchInput(TouchInput input) {
+    for(TouchInputs& points : _touchPoints){
+        if(points.getFingerId() == input.fingerId
```

Comment:
Looks a bit messy with nesting of ifs inside ifs with long unclear boolean values. 
Revisit to see if it can be simplified a bit.

Reply:
Simplified in e33004e